src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,891 @@
+'''
+
+from __future__ import division
+import logging
+
+from odemis import model
+from odemis.model import HwError
+
+import os
+import time
+import threading
+import socket
+import struct
+from _socket import timeout
+
+# Define constants
+DEFAULT_PORT = 6000
+DWELLTIME_RNG = (1e-9, 1000.0)
+
+'''
+Symphotime codes
+These values are defined in the symphotime documentation
+'''
+# Message Types
+PQ_MSGTYP_DATAFRAME_SRVREQUEST = 0x44  # server request message has to be answered with a server reply (i.e. 'd'-typed message)
+PQ_MSGTYP_DATAFRAME_SRVREPLY = 0x64  # answer on a server request (must not be answered)
+PQ_MSGTYP_DATAFRAME_SRVNACK = 0x78  # data frame sent by server (must not be answered)
+PQ_MSGTYP_ENCODED_STATUSMSG = 0x43  # status message has to be answered with a status reply (i.e. either 'c'-typed or 's'-typed message)
+PQ_MSGTYP_ENCODED_STATUSREPLY = 0x63  # answer on a status message (must not be answered)
+PQ_MSGTYP_EXPLAINED_STATUSMSG = 0x53    # status message enhanced by a free explaining text has to be answered with a status reply (i.e. either 'c'-typed or 's'-typed message)
+PQ_MSGTYP_EXPLAINED_STATUSREPLY = 0x73  # answer on a status message enhanced by a free explaining text (must not be answered)
+
+# Measurement types
+PQ_MEASTYPE_POINTMEAS = 0x00000000      # point measurement  
+PQ_MEASTYPE_IMAGESCAN = 0x00000001      # image scan  
+PQ_MEASTYPE_TEST_POINTMEAS = 0x00000080 # test run for a point measurement; no data will be stored in workspace  
+PQ_MEASTYPE_TEST_IMAGESCAN = 0x00000081 # test run for an image scan; no data will be stored in workspace  
+
+PQ_STOPREASON_CODE_CONTINUE_OK = 0      # no error detected  
+PQ_STOPREASON_CODE_FINISHED_OK = 1      # answer OK on finished server  
+PQ_STOPREASON_CODE_USER_BREAK = 2       # signalling an user break from client's site  
+PQ_STOPREASON_CODE_ERROR = -1           # signalling an error situation on client's site 
+
+# Types
+PQ_OPT_DATATYPE_FLOAT = 0x00            # floating point (4 byte, single precision)  
+PQ_OPT_DATATYPE_LONG  = 0x01            # integer (4 byte)  
+PQ_OPT_DATATYPE_ULONG = 0x02            #cardinal (unsigned integer, 4 byte)  
+PQ_OPT_DATATYPE_FLOATS_ARRAY = 0xF0     # floating point array (n × 4 byte, single precision)  
+PQ_OPT_DATATYPE_LONGS_ARRAY = 0xF1      # integer array (n × 4 byte)  
+PQ_OPT_DATATYPE_ULONGS_ARRAY = 0xF2     # cardinal array (n × 4 byte)  
+PQ_OPT_DATATYPE_FIXED_LENGTH_STRING = 0xFF  # character array (n byte string) 
+# it is recommended to be terminated with a NULL character   
+
+PQ_OPT_DATATYPE_NAME_MAXLEN = 30        # typical max length for string types
+
+# Error codes
+PQ_ERRCODE_NO_ERROR = 0
+PQ_ERRCODE_MEASUREMENT_READY = 1
+PQ_ERRCODE_USER_BREAK = 2
+
+# Header Constants
+T_REC_VERSION = '\x00\x02\x00\x01'      # Current record type version
+MAGIC = 'PQSPT'                         # magic string in header
+
+# dictionary of error codes and associated messages
+ERRCODE_DICT = {
```

Comment:
Please refrain from adding the type of a variable inside its name. (I know, that's exactly why the SymPhoTime documentation does, but it just makes names longer, and in most cases shouldn't be needed to read the code easily). So, I'd suggest to rename `ERRCODE_DICT` to `ERRCODES`.